src/harddisk/fat32.rs

```diff
@@ -0,0 +1,1554 @@
+		(&*value).try_into()
+	}
+}
+
+impl Into<[u8; 32]> for DirectoryEntry {
+	fn into(self) -> [u8; 32] {
+		match self {
+			DirectoryEntry::Standard {
+				file_name,
+				attributes,
+				first_cluster,
+				file_size,
+			} => {
+				let mut ret = [0; 32];
+
+				let mut name = SVec::<u8, 8>::new();
+				let mut ext = SVec::<u8, 3>::new();
+				let (bare_name, extension) = file_name.get_slice().split_last_2(&b'.');
+				for &b in bare_name {
+					name.push(b);
+				}
+				for _ in name.len()..name.capacity() {
+					name.push(b' ');
+				}
+				for &b in extension {
+					ext.push(b);
+				}
+				for _ in ext.len()..ext.capacity() {
+					ext.push(b' ');
+				}
+
+				ret[0..8].copy_from_slice(name.get_slice());
+				ret[8..11].copy_from_slice(ext.get_slice());
+				ret[11] = attributes;
+				ret[20..22].copy_from_slice(&((first_cluster >> 16) as u16).to_le_bytes());
+				ret[26..28].copy_from_slice(&(first_cluster as u16).to_le_bytes());
+				ret[28..32].copy_from_slice(&file_size.to_le_bytes());
+
+				ret
+			}
+			_ => unimplemented!(),
+		}
+	}
+}
+
+/// Initializes the FAT32 driver
+///
+/// # Safety
+///
+/// Requires partitions to have been initialized
+pub unsafe fn initialize() {
+	DRIVER.initialize();
+	// for file in driver.get_entries(b"").unwrap().get_slice() {
+	// 	println!(
+	// 		"{:12}  {:3}  {}",
+	// 		file.name.to_str(),
+	// 		if file.is_directory { "DIR" } else { "   " },
+	// 		file.size
+	// 	);
+	// }
+}
+
+/// Writes `data` to `path`
+pub unsafe fn write_file(path: Path, data: &[u8]) -> Result<(), FatError> {
+	DRIVER.write_file(path, data)
+}
+
+/// Puts the data from `path` in `buffer`
+///
+/// Returns size of file, succeed or fail.
+pub unsafe fn read_file(path: Path, buffer: &mut [u8]) -> Result<usize, FatError> {
+	DRIVER.read_file(path, buffer)
+}
+
+/// Get the `FileInfo` for the file at `path`
+pub unsafe fn get_file_info(path: Path) -> FileInfo {
+	DRIVER.get_entry_info(path).unwrap()
+}
+
+/// Lists all entries in `directory_path`
+pub unsafe fn list_entries(directory_path: Path) -> Result<Vec<FileInfo>, FatError> {
+	DRIVER.get_entries(directory_path)
+}
+
+/// `touch`
+///
+/// Creates an empty file at `path`
+pub unsafe fn create_empty_file(path: Path) -> Result<FileInfo, FatError> {
+	DRIVER.create_empty_file(path)
+}
+
+/// Used to split directories from each other in paths
+pub trait SplitLast<T>: Sized {
+	fn split_last_2(self, v: &T) -> (Self, Self);
+}
+
+impl<T: PartialEq> SplitLast<T> for &[T] {
+	/// Splits at the last occurence of `v`
+	///
+	/// # Example
+	/// ```
+	/// let s:SVec<char, 6> = SVec::new();
+	/// s.push("r")
+	/// s.push("o")
+	/// s.push("o")
+	/// s.push("t")
+	/// s.push("/")
+	/// s.push("d")
+	/// s.push("i")
+	/// s.push("r")
+	/// s.push("/")
+	/// s.push("f")
+	/// s.push("i")
+	/// s.push("l")
+	/// s.push("e")
+	///
+	/// asserteq!(("root".as_slice(), "dir/file".as_slice()), s.get_slice().split_last_2("/"));
```

Comment:
This could be re-done better now that we have (and this module requires) dynamic allocation